### ISSUE TITLE
Recognize CNAME cloaking 

### ISSUE THREAD

**Issue by eblocker-rulz** [eblocker/eblocker#150]
test.de is using Webtrekk with CNAME cloaking. images1.test.de actually belongs to Webtrekk.
I‘m not sure why/if this is blocked anyway (with pattern blocker Webtrack URLs could be recognized). PDFW shows pass on images1. Also test.de shows pass - even though contolbar indicates consent banner blocking. (But this might be another issue?)

**Comment by mainzelM** [comment]
Just for better understanding on my side: isn't this a problem with the blocking lists itself? I.e. shouldn't images1.test.de be on the list?

**Comment by eblocker-rulz** [comment]
Personally I wonder why the webtrekk script did not get blocked by the pattern blocker in this particular case. Privacy Badger and Adblock do recognize the tracker though.

Generally blocking „all“ tracking CNAMES via a list is practically impossible as the tracking domain is customer individual. But the IP and possibly the reverse lookup reveals all CNAMES pointing to the same tracker. 

So a better solution would be to add tracker IP addresses to the blocker list and make sure the DNS resolves first and blocker checks resolved IP. Possibly the reverse lookup of any IP in question would reveal the tracker domain as well which then can be matched against the pattern list.

**Comment by eblosug** [comment]
There is a paper [The CNAME of the Game: Large-scale Analysis of DNS-based Tracking Evasion](https://arxiv.org/pdf/2102.09301.pdf).

**Comment by eblosug** [comment]
I'm not sure how this can be implemented. Let's assume eBlocker's DNS is enabled and the pattern blocker is used:

* The browser tries to access `https://images1.test.de/368664393966025/wt` with many URL parameters (which is probably a tracking URL).
* The eBlocker DNS asks the Icapserver if the domain `images1.test.de` is blocked. The answer is no. Note that the pattern blocker is not involved here yet!
* The eBlocker DNS could parse the DNS response and find that `images1.test.de` has a CNAME entry `warentest-de01.webtrekk.net`.
* The pattern blocker tries to find a match for `https://images1.test.de/368664393966025/wt`. Probably there is no match.

What should the pattern blocker do now? Check all patterns again for `https://warentest-de01.webtrekk.net/368664393966025/wt`, assuming the eBlocker DNS has stored the CNAME relation somewhere?
